domain/schema_validator_test.go

```diff
@@ -67,9 +67,9 @@ func (*testSuite) TestSchemaValidator(c *C) {
 	<-oracleCh // Make sure that ts has timed out a lease.
 	time.Sleep(lease)
 	ts := <-oracleCh
-	valid = validator.Check(ts, item.schemaVer, []int64{10})
+	valid = validator.Check(ts, 0, []int64{10})
```

Comment:
Looks like the failing case returns `ResultSucc` instead of `ResultUnknown` in `schemaValidator::Check`, from its source code and log of the failing case(`schemaVer` is 1 while `s.latestSchemaVer` is 0), the function returns from line 181 or line 204. If we change the `schemaVer` from 1 to 0, it does not effect the code path of the failing case?

```golang
173 func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedTableIDs []int64) checkResult {
174     s.mux.RLock()
175     defer s.mux.RUnlock()
176     if !s.isStarted {
177         log.Infof("[domain-ddl] the schema validator stopped before checking")
178         return ResultUnknown
179     }
180     if s.lease == 0 {
181         return ResultSucc
182     }
183
184     // Schema changed, result decided by whether related tables change.
185     if schemaVer < s.latestSchemaVer {
186         // The DDL relatedTableIDs is empty.
187         if len(relatedTableIDs) == 0 {
188             log.Infof("[domain-ddl] the related table ID is empty, current schema version %d, latest schema version %d",
189                 schemaVer, s.latestSchemaVer)
190             return ResultFail
191         }
192
193         if s.isRelatedTablesChanged(schemaVer, relatedTableIDs) {
194             return ResultFail
195         }
196         return ResultSucc
197     }
198
199     // Schema unchanged, maybe success or the schema validator is unavailable.
200     t := oracle.GetTimeFromTS(txnTS)
201     if t.After(s.latestSchemaExpire) {
202         return ResultUnknown
203     }
204     return ResultSucc
205 }
```
From the log, `2157-07-06 11:58:30.888 +0000 UTC` is not `After` `2157-07-06 11:58:30.892 +0000 UTC`, is this the cause of the failing case?

Reply:
Yes.